src/lib/loaders/batchLoader.ts

```diff
@@ -59,39 +82,89 @@ interface BatchLoaderArgs {
    */
   singleLoader?: any
   multipleLoader: any
-  defaultResult?: any
 }
 
 export const batchLoader = ({
   singleLoader,
   multipleLoader,
-  defaultResult = null,
 }: BatchLoaderArgs) => {
   if (!ENABLE_RESOLVER_BATCHING) {
     return singleLoader ? singleLoader : multipleLoader
   }
-  const dl = new DataLoader(keys => {
-    let groupedKeys = groupKeys(keys as any)
-
-    return Promise.all(
-      groupedKeys.map(keys => {
-        if (keys.id.length === 1 && singleLoader) {
-          return singleLoader(keys.id[0])
-        } else {
-          return multipleLoader(keys)
-        }
-      })
-    ).then(data => {
-      const normalizedResults = data.map((queriedGroup, groupIndex) => {
-        return groupedKeys[groupIndex].id.map(
-          id => queriedGroup.find(r => r._id === id) || defaultResult
-        )
+  const dl = new DataLoader(
+    async (idWithParamsList: IdWithParams[]) => {
+      const [paramGroups, groupedParams] = groupByParams(idWithParamsList)
+      const data = await Promise.all(
+        groupedParams.map(batchParams).map(params_1 => {
```

Comment:
Why is this called `params_1`?

Reply:
Hrmm, I see, just to clarify to the reader that it’s different from `params_2` 🤔 

Ok, I have no better suggestions right now 🤷‍♂️

Reply:
Ummm... That... what. 

I'm not sure how that happened exactly... this seems like a vscode refactor gone wrong, but I _really_ don't remember seeing it. I'll update.